superset/sql_parse.py

```diff
@@ -332,8 +332,9 @@ def set_or_update_query_limit(self, new_limit: int) -> str:
                 break
         _, limit = statement.token_next(idx=limit_pos)
         # Override the limit only when it exceeds the configured value.
-        if limit.ttype == sqlparse.tokens.Literal.Number.Integer and new_limit < int(
-            limit.value
+        if force or (
+            limit.ttype == sqlparse.tokens.Literal.Number.Integer
+            and (new_limit < int(limit.value))
```

Comment:
We actually need to have this condition like we did the first time:

```suggestion
        if (
            limit.ttype == sqlparse.tokens.Literal.Number.Integer
            and (force or new_limit < int(limit.value))
```

Otherwise we'll change the next block (`elif limit.is_group:`), which TBH I don't understand.